### ISSUE TITLE
performance of batchSize > 1

### ISSUE THREAD

**Issue by ShaniGam** [junyanz/pytorch-CycleGAN-and-pix2pix#213]
I ran CycleGAN with batchSize = 2/4 and I got really bad results comparing to batchSize = 1, can you explain why it happens? Also tried to update the network only by the error of one of the images but it didn't help.

**Comment by meeio** [comment]
When batch_size = 1, it's instance_normalization. When batch_size > 1, it's batch_normalization. instance_normalization is better than batch_normalization for image2image transfer.

**Comment by ShaniGam** [comment]
I'm using the default which is instance_normalization (even when the batch > 1)

**Comment by taesung89** [comment]
What kind of bad results did you get? Did you see flipping of color? I sometimes see that the color flips depending on initialization. To resolve this, you can 

1. just start it multiple times until you get lucky and there is no color flipping. 
2. reduce `--init_variance`. 
3. add identity loss, at least in the first epoch of the training.

**Comment by HankYangg** [comment]
> When batch_size = 1, it's instance_normalization. When batch_size > 1, it's batch_normalization. instance_normalization is better than batch_normalization for image2image transfer.

Why is istance_normalization better than batch_normalization for image2image transfer?

**Comment by junyanz** [comment]
It seems that instancenorm works quite well for style transfer tasks in general.  Figure 1 in [AdaIN](https://arxiv.org/pdf/1703.06868.pdf) has provided a nice explanation.

**Comment by aamster** [comment]
I also noticed this. When training with batch_size=1, a domain validation metric evaluated on validation set improves steadily (and visual inspection of output shows improving quality in generated images). When batch_size > 1, the metric does not improve (and generated image quality does not improve). I am using instance norm.
